sql/src/test/java/io/crate/analyze/OptimizeTableAnalyzerTest.java

```diff
@@ -0,0 +1,219 @@
+
+    static class TestMetaDataModule extends MetaDataModule {
+
+        @Override
+        protected void bindSchemas() {
+            super.bindSchemas();
+            SchemaInfo schemaInfo = mock(SchemaInfo.class);
+            BlobTableInfo blobTableInfo = mock(BlobTableInfo.class);
+            when(blobTableInfo.ident()).thenReturn(TEST_BLOB_TABLE_IDENT);
+            when(schemaInfo.getTableInfo(TEST_BLOB_TABLE_IDENT.name())).thenReturn(blobTableInfo);
+            schemaBinder.addBinding(BlobSchemaInfo.NAME).toInstance(schemaInfo);
+
+            SchemaInfo docSchemaInfo = mock(SchemaInfo.class);
+            when(docSchemaInfo.getTableInfo(TEST_PARTITIONED_TABLE_IDENT.name()))
+                .thenReturn(TEST_PARTITIONED_TABLE_INFO);
+            when(docSchemaInfo.getTableInfo(USER_TABLE_IDENT.name())).thenReturn(USER_TABLE_INFO);
+
+            schemaBinder.addBinding(Schemas.DEFAULT_SCHEMA_NAME).toInstance(docSchemaInfo);
+        }
+    }
+
+    @Override
+    protected List<Module> getModules() {
+        List<Module> modules = super.getModules();
+        modules.addAll(Arrays.<Module>asList(
+            new MockedClusterServiceModule(),
+            new TestMetaDataModule(),
+            new MetaDataSysModule()
+        ));
+        return modules;
+    }
+
+    @Test
+    public void testOptimizeSystemTable() throws Exception {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("table 'sys.shards' cannot be optimized");
+        analyze("OPTIMIZE TABLE sys.shards");
+    }
+
+    @Test
+    public void testOptimizeBlobTable() throws Exception {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("table 'blob.blobs' cannot be optimized");
+        analyze("OPTIMIZE TABLE blob.blobs");
+    }
+
+    @Test
+    public void testOptimizeTable() throws Exception {
+        OptimizeTableAnalyzedStatement analysis = analyze("OPTIMIZE TABLE users");
+        assertThat(analysis.indexNames().size(), is(1));
+        assertThat(analysis.indexNames(), hasItem("users"));
+    }
+
+    @Test
+    public void testOptimizeTableWithParams() throws Exception {
+        OptimizeTableAnalyzedStatement analysis = analyze("OPTIMIZE TABLE users WITH (max_num_segments=2)");
+        assertThat(analysis.indexNames().size(), is(1));
+        assertThat(analysis.indexNames(), hasItem("users"));
+        analysis = analyze("OPTIMIZE TABLE users WITH (only_expunge_deletes=true)");
+        assertThat(analysis.indexNames().size(), is(1));
+        assertThat(analysis.indexNames(), hasItem("users"));
+        analysis = analyze("OPTIMIZE TABLE users WITH (flush=true)");
+        assertThat(analysis.indexNames().size(), is(1));
+        assertThat(analysis.indexNames(), hasItem("users"));
+    }
+
+    @Test
+    public void testOptimizeTableWithInvalidMaxNumSegmentsValueType() throws Exception {
```

Comment:
this test and below is testing the IntSetting or BooleanSetting and therefore should be tested in a unit test just to validate proper setting behaviour, pls remove or move to concrete setting test if not yet exist